packages/components/ContextualMenu/src/SubmenuItem.tsx

```diff
@@ -42,58 +42,88 @@ export const SubmenuItem = compose<SubmenuItemType>({
     const cmRef = useViewCommandFocus(componentRef);
 
     const onItemHoverIn = React.useCallback((e) => {
-      userProps.onHoverIn(e);
+      componentRef.current.focus();
+      userProps.onHoverIn && userProps.onHoverIn(e);
     }, []);
 
-    const onItemClick = React.useCallback(
+    const onHoverInDelay = Platform.select({
+      macos: 100,
+      default: 500, // win32
+    });
+
+    const onItemHoverOut = React.useCallback((e) => {
+      context?.dismissSubmenu && context.dismissSubmenu();
+      userProps.onHoverOut && userProps.onHoverOut(e);
+    }, []);
+
+    const onItemPress = React.useCallback(() => {
+      if (!disabled) {
+        onClick && onClick();
+        context?.onItemClick && context.onItemClick(itemKey);
+        context?.dismissSubmenu && context.dismissSubmenu();
+        context?.onDismissMenu();
+      }
+    }, [context, disabled, itemKey, onClick]);
+
+    const pressable = useAsPressable({
+      ...rest,
+      onPress: onItemPress,
+      onHoverIn: onItemHoverIn,
+      delayHoverIn: onHoverInDelay,
+      onHoverOut: onItemHoverOut,
+    });
+
+    /**
+     * GH #1267
+     * We want onMouseEnter to fire right away to set focus, and then Pressable's onHoverIn to fire after a delay to show the submenu.
+     * To achieve this, we override the onMouseEnter handler returned by useAsPressable, and replace it with our own. Inside our own
+     * onMouseEnter handler, we call useAsPressable's onMouseEnter handler, which incorporates the delay passed to delayHoverIn
+     * In the future, we can avoid needing to override onMouseEnter by handling submenu rendering internally rather than depending on the
+     * client to conditionally render it with onHoverIn.
+     */
+    const { onBlur, onMouseEnter, onMouseLeave, ...restPressableProps } = pressable.props;
+    const onMouseEnterModified = React.useCallback(
       (e) => {
-        if (!disabled) {
-          context?.onDismissMenu();
-          context?.dismissSubmenu && context.dismissSubmenu();
-          onClick ? onClick() : context?.onItemClick(itemKey);
-          e.stopPropagation();
-        }
+        componentRef.current.focus();
+        onMouseEnter && onMouseEnter(e);
       },
-      [context, disabled, itemKey, onClick],
+      [onMouseEnter],
```

Comment:
Does this not need to memoize on componentRef as well?

Reply:
My feeling is no, because `componentRef.current` should be properly updated..
demoing on a mutable object seems wrong 😅